ams/server/src/main/java/com/netease/arctic/server/table/TableMetrics.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.server.table;
+
+import static com.netease.arctic.ams.api.metrics.MetricDefine.defineGauge;
+
+import com.netease.arctic.ams.api.metrics.*;
+import com.netease.arctic.server.metrics.MetricRegistry;
+import com.netease.arctic.server.optimizing.OptimizingStatus;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+import java.util.List;
+
+public class TableMetrics {
+  /** Table is no need optimizing. */
+  public static final String STATE_IDLE = "idle";
+
+  /** Table is need optimizing, but waiting for resource */
+  public static final String STATE_PENDING = "pending";
+
+  /** Table is doing optimizing process planing. */
+  public static final String STATE_PLANING = "planing";
+
+  /** Table is executing optimizing process */
+  public static final String STATE_EXECUTING = "executing";
+
+  /** All optimizing process task is done, and process is committing. */
+  public static final String STATE_COMMITTING = "committing";
+
+  // table optimizing status duration metrics
+  public static final MetricDefine TABLE_OPTIMIZING_STATE_IDLE_DURATION =
+      defineGauge("table_optimizing_status_idle_duration_seconds")
+          .withDescription("Duration in seconds after table be in idle state")
```

Comment:
Considering that table status changes are sometimes relatively fast, such as small minor optimizations. Is it better to use milliseconds as the status metric unit?

Reply:
The cycle for table state transitions is at least in the order of seconds. The expected minimum duration for minor optimization is 3-5 seconds.

Reply:
Well, whether  the planning state duration is also on the seconds level  at least?

Reply:
You are right, the plan might be completed within 1 second at the shortest.